### PR TITLE
Remove encodeURI code

### LINKED ISSUE #381: Can't open PDF when filename or folder name contains a space or special character (' ', é, ...)
### :adhesive_bandage: temporary fix
:information_source: There is a temporary fix that takes replaces the pdfviewers code with that from PR #384. To do this replace
`apps/files_pdfviewer/js/files_pdfviewer-main.js` on your instance with [this file from that PR](https://raw.githubusercontent.com/nextcloud/files_pdfviewer/6d81ffbb65c3758bece144e0aff07b4a0ad20eef/js/files_pdfviewer-main.js). Clear your browser's cache and try to open pdfs.

That should fix the issue. Please leave feedback if that does or doesn't work for you. 

---

### Steps to reproduce
1. Create a New Folder
2. Add a PDF file to to the folder. PDF filename containing a space (e.g. name it "Juergen Klopp.pdf"). 
3. Open PDF file "Juergen Klopp.pdf"
4. It also won't work if the folder contains a space

### Expected behaviour
Open the PDF with PDF viewer

### Actual behaviour
PDF.js Version 2.5.207 (build: 0974d6052)
Nachricht: Missing PDF "https://domain.org/remote.php/dav/files/user/New/Juergen%2520Klopp.pdf".

## Server configuration detail

**Operating system:** Linux 5.4.0-65-generic #73-Ubuntu SMP Mon Jan 18 17:25:17 UTC 2021 x86_64

**Webserver:** Apache/2.4.41 (Ubuntu) (apache2handler)

**Database:** mysql 8.0.25

**PHP version:** 

8.0.5
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, json, Reflection, SPL, session, standard, sodium, apache2handler, mysqlnd, PDO, xml, apcu, bcmath, calendar, ctype, curl, dom, mbstring, FFI, fileinfo, ftp, gd, gettext, gmp, iconv, igbinary, imagick, intl, ldap, exif, mysqli, pdo_mysql, Phar, posix, readline, redis, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, xmlreader, xmlwriter, xsl, zip, Zend OPcache

**Nextcloud version:** 21.0.2 RC1 - 21.0.2.0

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** unknown

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.7.0
 - apporder: 0.12.0
 - calendar: 2.2.1
 - circles: 0.21.2
 - cloud_federation_api: 1.4.0
 - comments: 1.11.0
 - contacts: 3.5.1
 - contactsinteraction: 1.2.0
 - dav: 1.17.1
 - documentserver_community: 0.1.9
 - federatedfilesharing: 1.11.0
 - federation: 1.11.0
 - files: 1.16.0
 - files_external: 1.12.0
 - files_pdfviewer: 2.1.0
 - files_rightclick: 1.0.0
 - files_sharing: 1.13.1
 - files_trashbin: 1.11.0
 - files_versions: 1.14.0
 - files_videoplayer: 1.10.0
 - firstrunwizard: 2.10.0
 - forms: 2.2.4
 - issuetemplate: 0.7.0
 - logreader: 2.6.0
 - lookup_server_connector: 1.9.0
 - notes: 4.0.4
 - notifications: 2.9.0
 - oauth2: 1.9.0
 - onlyoffice: 7.0.2
 - passman: 2.3.7
 - password_policy: 1.11.0
 - photos: 1.3.0
 - privacy: 1.5.0
 - provisioning_api: 1.11.0
 - recommendations: 1.0.0
 - serverinfo: 1.11.0
 - settings: 1.3.0
 - sharebymail: 1.11.0
 - support: 1.4.0
 - survey_client: 1.9.0
 - systemtags: 1.11.0
 - tasks: 0.13.6
 - text: 3.2.0
 - theming: 1.12.0
 - theming_customcss: 1.8.0
 - twofactor_backupcodes: 1.10.0
 - updatenotification: 1.11.0
 - user_status: 1.1.1
 - viewer: 1.5.0
 - weather_status: 1.1.0
 - workflowengine: 2.3.0
Disabled:
 - activity
 - admin_audit
 - bruteforcesettings
 - dashboard
 - encryption
 - mail
 - nextcloud_announcements
 - user_ldap
 - workflow_pdf_converter

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "cloud.rekorder.org"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "21.0.2.0",
    "overwrite.cli.url": "http:\/\/cloud.rekorder.org",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "updater.release.channel": "stable",
    "memcache.local": "\\OC\\Memcache\\APCu",
    "maintenance": false,
    "data-fingerprint": "11bd4c43db46260d550ca15af9437513",
    "mail_smtpmode": "smtp",
    "mail_smtpsecure": "ssl",
    "mail_sendmailmode": "smtp",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauthtype": "LOGIN",
    "mail_smtpauth": 1,
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "465",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "theme": "",
    "loglevel": 2,
    "app_install_overwrite": [
        "issuetemplate"
    ]
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** 

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

## Client configuration

**Browser:** Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36

**Operating system:** 

## Logs

<details><summary>Web server error log</summary>

```
Insert your web server log here 
```
</details>

<details><summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

<details><summary>Browser log</summary>

Insert your browser log here, this could for example include:

	a) The javascript console log
	b) The network log
	c) ...
</details>

### PR DESCRIPTION
**Pull request by beardhatcode**
The errors that required encodeURIComponent in the past have been
mittigated by the following commit on nextcloud viewer.

https://github.com/nextcloud/viewer/pull/868/commits/02e1b1a993fb24f577f16ac22f816b89c578acf2

The code with hasScheme may be removed, viewer now returns a full
properly encoded URI thanks to the work done in nextcloud-router

fixes #381

---

:information_source: You can teporarily apply this patch by replacing
`apps/files_pdfviewer/js/files_pdfviewer-main.js` on your instance with [this file from this PR](https://raw.githubusercontent.com/nextcloud/files_pdfviewer/6d81ffbb65c3758bece144e0aff07b4a0ad20eef/js/files_pdfviewer-main.js). Clear your browser's cache and try to open pdfs.

That should fix the issue. Please leave feedback if that does or doesn't work for you.

### CODE REVIEW

**Comment by beardhatcode** [comment]
I tested this for local files with the name "2020_opgave 🪓 NL .pdf"

**Comment by beardhatcode** [comment]
I was wondering if there is a way to ship updates to this app outside a release of Nextcloud, and if that is something that would be appropriate for this issue?

**Comment by jacotec** [comment]
This bug breaks the PDF viewer in all NC 21.0.2 installations. I guess one of the most used features. I've applied the hotfix from @beardhatcode manually, but in the interest of all NC users this update should be pushed ASAP and not only with the next NC release. Just my 2 cent.

However I wonder how this did pass release testing ...

**Comment by acsfer** [comment]
@MorrisJobke maybe this is something you can make a quick release? 🤔

**Comment by MorrisJobke** [comment]
cc @LukasReschke

**Review by LukasReschke — APPROVED** [review]
Makes sense. Thanks for putting up a diff, much appreciated.

**Review by skjnldsv — APPROVED** [review]

File: src/views/PDFView.vue [hunk 1]
```diff
@@ -33,41 +33,9 @@ export default {
...
-					encodedDavPath += '/' + encodeURIComponent(pathSections[i])
-				}
-			}
-
-			if (hasScheme) {
-				encodedDavPath = pathSections[0] + '//' + pathSections[2] + encodedDavPath
-			}
-
-			return encodedDavPath
-		},
```
Comment (skjnldsv) [inline src/views/PDFView.vue:70]: Good rid of this legacy code! :broom:

**Comment by livier** [comment]
The patch worked here, thanks.

**Comment by jdaviescoates** [comment]
Many thanks of this fix, but how long will we need to wait to this get into an actual Nextcloud release? I'm fairly flabbergasted that 21.0.2 was released without basic PDF support! Really very disappointing :disappointed:

**Comment by LukasReschke** [comment]
> Many thanks of this fix, but how long will we need to wait to this get into an actual Nextcloud release? 

The next regular release is scheduled for 2021-07-01 (https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule)

**Pull request merged**